consul/structs/prepared_query.go

```diff
@@ -173,6 +178,12 @@ type PreparedQueryExecuteRequest struct {
 	// Limit will trim the resulting list down to the given limit.
 	Limit int
 
+	// Origin is used to carry around a reference to the node which
+	// is executing the query on behalf of the client. It is taken
+	// as a QuerySource so that it can be used directly for queries
+	// relating to the agent servicing the request.
+	Origin QuerySource
```

Comment:
Did you take a look at the existing `Source` parameter right after this one? I think it was originally intended to do what `Origin` does.